plugins/linux-capture/xcomposite-input.c

```diff
@@ -615,7 +615,8 @@ void watcher_unregister(xcb_connection_t *conn, struct xcompcap *s)
 		}
 	}
 
-	da_erase(watcher_registry, idx);
+	if (idx != DARRAY_INVALID)
+		da_erase(watcher_registry, idx);
```

Comment:
Hm, do you not need to early return here even? If the window is not in `watcher_registry`, there's no need to iterate over it yet another time. Early returning would need to unlock `watcher_lock` though.

Reply:
Is the intent to set `idx = i` before the `break` in the loop above? As I'm reading it, `idx` will always eq `DARRAY_INVALID` here.

Reply:
Yes that was the original issue, it appears I completely forgot to add in the assignments to `idx` in both places. Likely the size of the watcher table just ended up growing and growing, but double checking with the assignments I forgot in place things look good.